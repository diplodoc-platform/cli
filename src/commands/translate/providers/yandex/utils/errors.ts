import {TranslateError} from '../../../utils';

export class RequestError extends TranslateError {
    static canRetry(error: any) {
        if (error instanceof RequestError) {
            switch (true) {
                case error.status === 429:
                    return true;
                case error.status === 500:
                    return true;
                case error.status === 503:
                    return true;
                case error.status === 504:
                    return true;
                default:
                    return false;
            }
        }

        return false;
    }

    status: number;

    constructor(
        status: number,
        statusText: string,
        info: {code?: number; message?: string; fatal?: boolean} = {},
    ) {
        super(`${statusText}\n${info.message || ''}`, 'REQUEST_ERROR', info.fatal);

        this.status = status;
    }
}

const INACTIVE_CLOUD = /^The cloud .*? is inactive/;
const WRONG_APIKEY = /^Unknown api key/;
const WRONG_TOKEN = /^The token is invalid/;
const EXPIRED_TOKEN = /^The token has expired/;

export class AuthError extends TranslateError {
    static is(message: string) {
        return Boolean(AuthError.reason(message));
    }

    static reason(message: string) {
        switch (true) {
            case INACTIVE_CLOUD.test(message):
                return 'INACTIVE_CLOUD';
            case WRONG_APIKEY.test(message):
                return 'WRONG_APIKEY';
            case WRONG_TOKEN.test(message):
                return 'WRONG_TOKEN';
            case EXPIRED_TOKEN.test(message):
                return 'EXPIRED_TOKEN';
            default:
                return null;
        }
    }

    constructor(message: string) {
        super(message, AuthError.reason(message) || 'AUTH_ERROR', true);
    }
}

const LIMIT_EXCEED_RX = /^limit on units was exceeded. (.*)$/;

export class LimitExceed extends TranslateError {
    static is(message: string) {
        return Boolean(LIMIT_EXCEED_RX.test(message));
    }

    constructor(message: string) {
        const [, desc] = LIMIT_EXCEED_RX.exec(message) || [];
        super(desc, 'TRANSLATE_LIMIT_EXCEED', true);
    }
}
