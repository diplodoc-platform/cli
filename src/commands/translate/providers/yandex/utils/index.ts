export {LimitExceed, RequestError, AuthError} from './errors';

export class Defer<T = string> {
    resolve!: (text: T) => void;

    reject!: (error: any) => void;

    promise: Promise<T>;

    constructor() {
        this.promise = new Promise((resolve, reject) => {
            this.resolve = resolve;
            this.reject = reject;
        });
    }
}

export function bytes(texts: string[]) {
    return texts.reduce((sum, text) => sum + text.length, 0);
}
