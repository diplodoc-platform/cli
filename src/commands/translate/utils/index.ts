export type {Locale} from './config';
export {resolveSchemas, FileLoader} from './fs';
export {extract, compose} from './translate';
export {resolveSource, resolveTargets, resolveFiles, resolveVars} from './config';
export {
    TranslateError,
    ExtractError,
    ComposeError,
    SkipTranslation,
    EmptyTokensError,
} from './errors';
