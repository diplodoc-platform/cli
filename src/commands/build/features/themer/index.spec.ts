import {compareDirectories} from './__tests__';
import {getTestPaths, runYfmDocs} from '~/../tests/utils';
import {describe, it, expect} from 'vitest';
import {join} from 'node:path';
import {createCSS, createTheme} from './utils';
import {Theme, ThemeConfig} from './types';

const generateMapTestTemplate = (
    testTitle: string,
    testRootPath: string,
    {md2md = true, md2html = true, args = ''},
) => {
    it(testTitle, () => {
        const {inputPath, outputPath} = getTestPaths(join(__dirname, testRootPath));
        runYfmDocs(inputPath, outputPath, {md2md, md2html, args});
        compareDirectories(outputPath);
    });
};

describe('Build themer feature', () => {
    describe('Apply theme', () => {
        generateMapTestTemplate('md2md with theme yaml', '__tests__/mocks/md2md-with-theme-yaml', {
            md2html: false,
        });

        generateMapTestTemplate(
            'md2html with theme yaml',
            '__tests__/mocks/md2html-with-theme-yaml',
            {
                md2md: false,
            },
        );
    });

    describe('Create theme', () => {
        it('Light and dark themes', () => {
            const config: ThemeConfig = {
                light: {
                    'text-brand': 'rgb(0, 255, 0)',
                    'base-misc-light': '#F80000',
                },
                dark: {
                    'note-info-background': 'rgb(0 0 255)',
                    'note-tip-background': 'rgb(106,90,205)',
                },
            };
            const theme = createTheme(config);

            const expectedTheme: Theme = {
                light: {
                    colors: {'text-brand': 'rgb(0, 255, 0)', 'base-misc-light': '#F80000'},
                },
                dark: {
                    colors: {
                        'note-info-background': 'rgb(0 0 255)',
                        'note-tip-background': 'rgb(106,90,205)',
                    },
                },
            };
            expect(theme).toEqual(expectedTheme);
        });

        it('Brand colors from general', () => {
            const config: ThemeConfig = {
                'base-brand': 'rgb(94 33 41)',
            };
            const theme = createTheme(config);

            const expectedTheme: Theme = {
                base: {colors: {'base-brand': 'rgb(94 33 41)'}},
                light: {
                    colors: {
                        'base-brand': 'rgb(94 33 41)',
                        'base-background': 'rgb(255,255,255)',
                        'base-brand-hover': 'var(--g-color-private-brand-650-solid)',
                        'base-selection': 'var(--g-color-private-brand-150)',
                        'base-selection-hover': 'var(--g-color-private-brand-300)',
                        'text-link': 'var(--g-color-private-brand-700-solid)',
                        'text-link-hover': 'var(--g-color-private-brand-850-solid)',
                        'text-brand': 'var(--g-color-private-brand-700-solid)',
                        'text-brand-heavy': 'var(--g-color-private-brand-850-solid)',
                        'line-brand': 'var(--g-color-private-brand-550-solid)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                        '150': 'rgb(94 33 41 / 0.2)',
                        '200': 'rgb(94 33 41 / 0.3)',
                        '250': 'rgb(94 33 41 / 0.4)',
                        '300': 'rgb(94 33 41 / 0.5)',
                        '350': 'rgb(94 33 41 / 0.6)',
                        '400': 'rgb(94 33 41 / 0.7)',
                        '450': 'rgb(94 33 41 / 0.8)',
                        '500': 'rgb(94 33 41 / 0.9)',
                        '550-solid': 'rgb(94 33 41)',
                        '1000-solid': 'rgb(52 42 50)',
                        '950-solid': 'rgb(55 42 49)',
                        '900-solid': 'rgb(60 41 48)',
                        '850-solid': 'rgb(65 40 47)',
                        '800-solid': 'rgb(70 39 46)',
                        '750-solid': 'rgb(74 37 45)',
                        '700-solid': 'rgb(79 36 44)',
                        '650-solid': 'rgb(84 35 43)',
                        '600-solid': 'rgb(89 34 42)',
                        '500-solid': 'rgb(89 34 42)',
                        '450-solid': 'rgb(84 35 43)',
                        '400-solid': 'rgb(79 36 44)',
                        '350-solid': 'rgb(74 37 45)',
                        '300-solid': 'rgb(70 39 46)',
                        '250-solid': 'rgb(65 40 47)',
                        '200-solid': 'rgb(60 41 48)',
                        '150-solid': 'rgb(55 42 49)',
                        '100-solid': 'rgb(52 42 50)',
                        '50-solid': 'rgb(50 43 50)',
                    },
                },
                dark: {
                    colors: {
                        'base-brand': 'rgb(94 33 41)',
                        'base-background': 'rgb(45, 44, 51)',
                        'base-brand-hover': 'var(--g-color-private-brand-650-solid)',
                        'base-selection': 'var(--g-color-private-brand-150)',
                        'base-selection-hover': 'var(--g-color-private-brand-300)',
                        'text-link': 'var(--g-color-private-brand-600-solid)',
                        'text-link-hover': 'var(--g-color-private-brand-850-solid)',
                        'text-brand': 'var(--g-color-private-brand-600-solid)',
                        'text-brand-heavy': 'var(--g-color-private-brand-850-solid)',
                        'line-brand': 'var(--g-color-private-brand-550-solid)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                        '150': 'rgb(94 33 41 / 0.2)',
                        '200': 'rgb(94 33 41 / 0.3)',
                        '250': 'rgb(94 33 41 / 0.4)',
                        '300': 'rgb(94 33 41 / 0.5)',
                        '350': 'rgb(94 33 41 / 0.6)',
                        '400': 'rgb(94 33 41 / 0.7)',
                        '450': 'rgb(94 33 41 / 0.8)',
                        '500': 'rgb(94 33 41 / 0.9)',
                        '550-solid': 'rgb(94 33 41)',
                        '1000-solid': 'rgb(231 222 223)',
                        '950-solid': 'rgb(223 211 212)',
                        '900-solid': 'rgb(207 188 191)',
                        '850-solid': 'rgb(191 166 169)',
                        '800-solid': 'rgb(175 144 148)',
                        '750-solid': 'rgb(158 122 127)',
                        '700-solid': 'rgb(142 100 105)',
                        '650-solid': 'rgb(126 77 84)',
                        '600-solid': 'rgb(110 55 62)',
                        '500-solid': 'rgb(110 55 62)',
                        '450-solid': 'rgb(126 77 84)',
                        '400-solid': 'rgb(142 100 105)',
                        '350-solid': 'rgb(158 122 127)',
                        '300-solid': 'rgb(175 144 148)',
                        '250-solid': 'rgb(191 166 169)',
                        '200-solid': 'rgb(207 188 191)',
                        '150-solid': 'rgb(223 211 212)',
                        '100-solid': 'rgb(231 222 223)',
                        '50-solid': 'rgb(239 233 234)',
                    },
                },
            };
            expect(theme).toEqual(expectedTheme);
        });
    });

    describe('create css', () => {
        it('Create base css', () => {
            const Theme: Theme = {
                base: {
                    colors: {
                        'base-brand': 'rgb(94 33 41)',
                        'note-tip-background': 'rgb(106, 90, 205)',
                    },
                },
            };
            expect(createCSS(Theme)).toMatchSnapshot();
        });

        it('Create light css', () => {
            const Theme: Theme = {
                light: {
                    colors: {
                        'base-brand': 'rgb(48, 149, 232)',
                        'base-background': 'rgb(255,255,255)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                    },
                },
            };
            expect(createCSS(Theme)).toMatchSnapshot();
        });

        it('Create dark css', () => {
            const Theme: Theme = {
                light: {
                    colors: {
                        'base-brand': 'rgb(220, 48, 232)',
                        'base-background': 'rgb(255,255,255)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                    },
                },
            };
            expect(createCSS(Theme)).toMatchSnapshot();
        });

        it('Create full css', () => {
            const Theme: Theme = {
                base: {
                    colors: {
                        'base-brand': 'rgb(94 33 41)',
                        'note-tip-background': 'rgb(106, 90, 205)',
                    },
                },
                light: {
                    colors: {
                        'base-brand': 'rgb(48, 149, 232)',
                        'base-background': 'rgb(255,255,255)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                    },
                },
                dark: {
                    colors: {
                        'base-brand': 'rgb(58, 228, 143)',
                        'base-background': 'rgb(45, 44, 51)',
                    },
                    palette: {
                        '50': 'rgb(94 33 41 / 0.1)',
                        '100': 'rgb(94 33 41 / 0.15)',
                    },
                },
            };
            expect(createCSS(Theme)).toMatchSnapshot();
        });

        it('Create empty css', () => {
            const Theme: Theme = {};
            expect(createCSS(Theme)).toMatch('');
        });
    });
});
