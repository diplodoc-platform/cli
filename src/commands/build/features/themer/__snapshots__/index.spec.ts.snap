// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Build themer feature > Apply theme > md2html with theme yaml 1`] = `
"[
  ".yfm",
  "_assets/style/_theme.css",
  "_bundle/search-async-0",
  "_bundle/934.js",
  "_bundle/app-css-1",
  "_bundle/app-js-1",
  "_bundle/search-css-2",
  "_bundle/latex-extension.css",
  "_bundle/latex-extension.js",
  "_bundle/mermaid-extension.js",
  "_bundle/search-js-0",
  "_bundle/search-js-1",
  "_bundle/search/index.js",
  "_bundle/search/langs/ar.js",
  "_bundle/search/langs/da.js",
  "_bundle/search/langs/de.js",
  "_bundle/search/langs/du.js",
  "_bundle/search/langs/el.js",
  "_bundle/search/langs/es.js",
  "_bundle/search/langs/fi.js",
  "_bundle/search/langs/fr.js",
  "_bundle/search/langs/he.js",
  "_bundle/search/langs/hi.js",
  "_bundle/search/langs/hu.js",
  "_bundle/search/langs/hy.js",
  "_bundle/search/langs/index.js",
  "_bundle/search/langs/it.js",
  "_bundle/search/langs/ja.js",
  "_bundle/search/langs/jp.js",
  "_bundle/search/langs/kn.js",
  "_bundle/search/langs/ko.js",
  "_bundle/search/langs/nl.js",
  "_bundle/search/langs/no.js",
  "_bundle/search/langs/pt.js",
  "_bundle/search/langs/ro.js",
  "_bundle/search/langs/ru.js",
  "_bundle/search/langs/sa.js",
  "_bundle/search/langs/sv.js",
  "_bundle/search/langs/ta.js",
  "_bundle/search/langs/te.js",
  "_bundle/search/langs/th.js",
  "_bundle/search/langs/tr.js",
  "_bundle/search/langs/vi.js",
  "_bundle/search-css-0",
  "_bundle/search-js-2",
  "_bundle/search-css-1",
  "index.html",
  "page.html",
  "project/config.html",
  "toc.js"
]"
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 2`] = `
"resources:
  style:
    - _assets/style/test.css
  script:
    - _assets/script/test1.js
"
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 3`] = `
".g-root {
    --g-color-base-brand: rgb(94 33 41);
}

.g-root_theme_light {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-private-brand-150: rgb(94 33 41 / 0.2);
    --g-color-private-brand-200: rgb(94 33 41 / 0.3);
    --g-color-private-brand-250: rgb(94 33 41 / 0.4);
    --g-color-private-brand-300: rgb(94 33 41 / 0.5);
    --g-color-private-brand-350: rgb(94 33 41 / 0.6);
    --g-color-private-brand-400: rgb(94 33 41 / 0.7);
    --g-color-private-brand-450: rgb(94 33 41 / 0.8);
    --g-color-private-brand-500: rgb(94 33 41 / 0.9);
    --g-color-private-brand-550-solid: rgb(94 33 41);
    --g-color-private-brand-1000-solid: rgb(52 42 50);
    --g-color-private-brand-950-solid: rgb(55 42 49);
    --g-color-private-brand-900-solid: rgb(60 41 48);
    --g-color-private-brand-850-solid: rgb(65 40 47);
    --g-color-private-brand-800-solid: rgb(70 39 46);
    --g-color-private-brand-750-solid: rgb(74 37 45);
    --g-color-private-brand-700-solid: rgb(79 36 44);
    --g-color-private-brand-650-solid: rgb(84 35 43);
    --g-color-private-brand-600-solid: rgb(89 34 42);
    --g-color-private-brand-500-solid: rgb(89 34 42);
    --g-color-private-brand-450-solid: rgb(84 35 43);
    --g-color-private-brand-400-solid: rgb(79 36 44);
    --g-color-private-brand-350-solid: rgb(74 37 45);
    --g-color-private-brand-300-solid: rgb(70 39 46);
    --g-color-private-brand-250-solid: rgb(65 40 47);
    --g-color-private-brand-200-solid: rgb(60 41 48);
    --g-color-private-brand-150-solid: rgb(55 42 49);
    --g-color-private-brand-100-solid: rgb(52 42 50);
    --g-color-private-brand-50-solid: rgb(50 43 50);
    --g-color-base-brand: rgb(94 33 41);
    --g-color-base-background: rgb(255,255,255);
    --g-color-base-brand-hover: var(--g-color-private-brand-650-solid);
    --g-color-base-selection: var(--g-color-private-brand-150);
    --g-color-base-selection-hover: var(--g-color-private-brand-300);
    --g-color-text-link: var(--g-color-private-brand-700-solid);
    --g-color-text-link-hover: var(--g-color-private-brand-850-solid);
    --g-color-text-brand: rgb(0, 255, 0);
    --g-color-text-brand-heavy: var(--g-color-private-brand-850-solid);
    --g-color-line-brand: var(--g-color-private-brand-550-solid);
}

.g-root_theme_dark {
    --g-color-private-brand-50: rgb(255 255 0 / 0.1);
    --g-color-private-brand-100: rgb(255 255 0 / 0.15);
    --g-color-private-brand-150: rgb(255 255 0 / 0.2);
    --g-color-private-brand-200: rgb(255 255 0 / 0.3);
    --g-color-private-brand-250: rgb(255 255 0 / 0.4);
    --g-color-private-brand-300: rgb(255 255 0 / 0.5);
    --g-color-private-brand-350: rgb(255 255 0 / 0.6);
    --g-color-private-brand-400: rgb(255 255 0 / 0.7);
    --g-color-private-brand-450: rgb(255 255 0 / 0.8);
    --g-color-private-brand-500: rgb(255 255 0 / 0.9);
    --g-color-private-brand-550-solid: rgb(255 255 0);
    --g-color-private-brand-1000-solid: rgb(255 255 217);
    --g-color-private-brand-950-solid: rgb(255 255 204);
    --g-color-private-brand-900-solid: rgb(255 255 179);
    --g-color-private-brand-850-solid: rgb(255 255 153);
    --g-color-private-brand-800-solid: rgb(255 255 128);
    --g-color-private-brand-750-solid: rgb(255 255 102);
    --g-color-private-brand-700-solid: rgb(255 255 77);
    --g-color-private-brand-650-solid: rgb(255 255 51);
    --g-color-private-brand-600-solid: rgb(255 255 25);
    --g-color-private-brand-500-solid: rgb(255 255 25);
    --g-color-private-brand-450-solid: rgb(255 255 51);
    --g-color-private-brand-400-solid: rgb(255 255 77);
    --g-color-private-brand-350-solid: rgb(255 255 102);
    --g-color-private-brand-300-solid: rgb(255 255 128);
    --g-color-private-brand-250-solid: rgb(255 255 153);
    --g-color-private-brand-200-solid: rgb(255 255 179);
    --g-color-private-brand-150-solid: rgb(255 255 204);
    --g-color-private-brand-100-solid: rgb(255 255 217);
    --g-color-private-brand-50-solid: rgb(255 255 230);
    --g-color-base-brand: yellow;
    --g-color-base-background: rgb(45, 44, 51);
    --g-color-base-brand-hover: var(--g-color-private-brand-650-solid);
    --g-color-base-selection: var(--g-color-private-brand-150);
    --g-color-base-selection-hover: var(--g-color-private-brand-300);
    --g-color-text-link: var(--g-color-private-brand-600-solid);
    --g-color-text-link-hover: var(--g-color-private-brand-850-solid);
    --g-color-text-brand: var(--g-color-private-brand-600-solid);
    --g-color-text-brand-heavy: var(--g-color-private-brand-850-solid);
    --g-color-line-brand: var(--g-color-private-brand-550-solid);
}

.g-root_theme_dark .yfm {
    --yfm-color-note-info-background: blue;
}

"
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 4`] = `
"
        <!DOCTYPE html>
        <html lang="ru" dir="ltr">
            <head>
                <meta charset="utf-8">
                <base href="./" />
                <meta name="title" content="Documentation"  />
<meta name="noIndex" content="true"  />

                
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <title>Documentation | Documentation</title>
                <style type="text/css">
                    body {
                        height: 100vh;
                    }
                </style>
                <link type="text/css" rel="stylesheet" href="_bundle/search-css-0" />
<link type="text/css" rel="stylesheet" href="_bundle/app-css-1" />
                
                <link rel="stylesheet" type="text/css" href="_assets/style/_theme.css" id="custom-style">
            </head>
            <body class="g-root g-root_theme_light">
                <div id="root"></div>
                <script type="application/javascript">
                   window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"data":{"title":"Documentation","description":"","meta":{"title":"Documentation","noIndex":true,"metadata":[],"style":["_assets/style/_theme.css"],"script":[],"csp":[]},"links":[{"title":"Getting started with Documentation","description":"This guide will show you the basics of working with Documentation","href":"page.html"}]},"meta":{"title":"Documentation","noIndex":true,"metadata":[],"style":["_assets/style/_theme.css"],"script":[],"csp":[]},"title":"Documentation","leading":true},"router":{"pathname":"index","depth":1},"lang":"ru","langs":["ru"]};
                </script>
                <script src="toc.js" type="application/javascript"></script>
                
                <script type="application/javascript" src="_bundle/search-js-0"></script>
<script type="application/javascript" src="_bundle/app-js-1"></script>
<script type="application/javascript" src="_bundle/search-js-2"></script>
            </body>
        </html>
    "
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 5`] = `
"
        <!DOCTYPE html>
        <html lang="ru" dir="ltr">
            <head>
                <meta charset="utf-8">
                <base href="./" />
                <meta name="yfm" content="builder"  />

                
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <title>Documentation</title>
                <style type="text/css">
                    body {
                        height: 100vh;
                    }
                </style>
                <link type="text/css" rel="stylesheet" href="_bundle/search-css-0" />
<link type="text/css" rel="stylesheet" href="_bundle/app-css-1" />
                
                <link rel="stylesheet" type="text/css" href="_assets/style/_theme.css" id="custom-style">
            </head>
            <body class="g-root g-root_theme_light">
                <div id="root"></div>
                <script type="application/javascript">
                   window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{"metadata":[{"name":"yfm","content":"builder"}],"style":["_assets/style/_theme.css"],"script":[],"csp":[]},"assets":[],"headings":[],"title":"","includes":[],"html":"<p>Lorem</p>/n","leading":false},"router":{"pathname":"page","depth":1},"lang":"ru","langs":["ru"]};
                </script>
                <script src="toc.js" type="application/javascript"></script>
                
                <script type="application/javascript" src="_bundle/search-js-0"></script>
<script type="application/javascript" src="_bundle/app-js-1"></script>
<script type="application/javascript" src="_bundle/search-js-2"></script>
            </body>
        </html>
    "
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 6`] = `
"
        <!DOCTYPE html>
        <html lang="ru" dir="ltr">
            <head>
                <meta charset="utf-8">
                <base href="../" />
                
                
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <title>Documentation</title>
                <style type="text/css">
                    body {
                        height: 100vh;
                    }
                </style>
                <link type="text/css" rel="stylesheet" href="_bundle/search-css-0" />
<link type="text/css" rel="stylesheet" href="_bundle/app-css-1" />
                
                <link rel="stylesheet" type="text/css" href="_assets/style/_theme.css" id="custom-style">
            </head>
            <body class="g-root g-root_theme_light">
                <div id="root"></div>
                <script type="application/javascript">
                   window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{"metadata":[],"style":["_assets/style/_theme.css"],"script":[],"csp":[]},"assets":[],"headings":[],"title":"","includes":[],"html":"<p>Lorem</p>/n","leading":false},"router":{"pathname":"project/config","depth":2},"lang":"ru","langs":["ru"]};
                </script>
                <script src="toc.js" type="application/javascript"></script>
                
                <script type="application/javascript" src="_bundle/search-js-0"></script>
<script type="application/javascript" src="_bundle/app-js-1"></script>
<script type="application/javascript" src="_bundle/search-js-2"></script>
            </body>
        </html>
    "
`;

exports[`Build themer feature > Apply theme > md2html with theme yaml 7`] = `"window.__DATA__.data.toc = {"title":"Documentation","href":"index.html","items":[{"name":"Documentation","href":"page.html","id":"UUID"},{"name":"Config","href":"project/config.html","id":"UUID"}],"id":"UUID"};"`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 1`] = `
"[
  ".yfm",
  "_assets/style/_theme.css",
  "index.yaml",
  "page.md",
  "project/config.md",
  "toc.yaml"
]"
`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 2`] = `
"resources:
  style:
    - _assets/style/test.css
  script:
    - _assets/script/test1.js
"
`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 3`] = `
".g-root {
    --g-color-base-brand: rgb(94 33 41);
}

.g-root_theme_light {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-private-brand-150: rgb(94 33 41 / 0.2);
    --g-color-private-brand-200: rgb(94 33 41 / 0.3);
    --g-color-private-brand-250: rgb(94 33 41 / 0.4);
    --g-color-private-brand-300: rgb(94 33 41 / 0.5);
    --g-color-private-brand-350: rgb(94 33 41 / 0.6);
    --g-color-private-brand-400: rgb(94 33 41 / 0.7);
    --g-color-private-brand-450: rgb(94 33 41 / 0.8);
    --g-color-private-brand-500: rgb(94 33 41 / 0.9);
    --g-color-private-brand-550-solid: rgb(94 33 41);
    --g-color-private-brand-1000-solid: rgb(52 42 50);
    --g-color-private-brand-950-solid: rgb(55 42 49);
    --g-color-private-brand-900-solid: rgb(60 41 48);
    --g-color-private-brand-850-solid: rgb(65 40 47);
    --g-color-private-brand-800-solid: rgb(70 39 46);
    --g-color-private-brand-750-solid: rgb(74 37 45);
    --g-color-private-brand-700-solid: rgb(79 36 44);
    --g-color-private-brand-650-solid: rgb(84 35 43);
    --g-color-private-brand-600-solid: rgb(89 34 42);
    --g-color-private-brand-500-solid: rgb(89 34 42);
    --g-color-private-brand-450-solid: rgb(84 35 43);
    --g-color-private-brand-400-solid: rgb(79 36 44);
    --g-color-private-brand-350-solid: rgb(74 37 45);
    --g-color-private-brand-300-solid: rgb(70 39 46);
    --g-color-private-brand-250-solid: rgb(65 40 47);
    --g-color-private-brand-200-solid: rgb(60 41 48);
    --g-color-private-brand-150-solid: rgb(55 42 49);
    --g-color-private-brand-100-solid: rgb(52 42 50);
    --g-color-private-brand-50-solid: rgb(50 43 50);
    --g-color-base-brand: rgb(94 33 41);
    --g-color-base-background: rgb(255,255,255);
    --g-color-base-brand-hover: var(--g-color-private-brand-650-solid);
    --g-color-base-selection: var(--g-color-private-brand-150);
    --g-color-base-selection-hover: var(--g-color-private-brand-300);
    --g-color-text-link: var(--g-color-private-brand-700-solid);
    --g-color-text-link-hover: var(--g-color-private-brand-850-solid);
    --g-color-text-brand: rgb(0, 255, 0);
    --g-color-text-brand-heavy: var(--g-color-private-brand-850-solid);
    --g-color-line-brand: var(--g-color-private-brand-550-solid);
}

.g-root_theme_dark {
    --g-color-private-brand-50: rgb(255 255 0 / 0.1);
    --g-color-private-brand-100: rgb(255 255 0 / 0.15);
    --g-color-private-brand-150: rgb(255 255 0 / 0.2);
    --g-color-private-brand-200: rgb(255 255 0 / 0.3);
    --g-color-private-brand-250: rgb(255 255 0 / 0.4);
    --g-color-private-brand-300: rgb(255 255 0 / 0.5);
    --g-color-private-brand-350: rgb(255 255 0 / 0.6);
    --g-color-private-brand-400: rgb(255 255 0 / 0.7);
    --g-color-private-brand-450: rgb(255 255 0 / 0.8);
    --g-color-private-brand-500: rgb(255 255 0 / 0.9);
    --g-color-private-brand-550-solid: rgb(255 255 0);
    --g-color-private-brand-1000-solid: rgb(255 255 217);
    --g-color-private-brand-950-solid: rgb(255 255 204);
    --g-color-private-brand-900-solid: rgb(255 255 179);
    --g-color-private-brand-850-solid: rgb(255 255 153);
    --g-color-private-brand-800-solid: rgb(255 255 128);
    --g-color-private-brand-750-solid: rgb(255 255 102);
    --g-color-private-brand-700-solid: rgb(255 255 77);
    --g-color-private-brand-650-solid: rgb(255 255 51);
    --g-color-private-brand-600-solid: rgb(255 255 25);
    --g-color-private-brand-500-solid: rgb(255 255 25);
    --g-color-private-brand-450-solid: rgb(255 255 51);
    --g-color-private-brand-400-solid: rgb(255 255 77);
    --g-color-private-brand-350-solid: rgb(255 255 102);
    --g-color-private-brand-300-solid: rgb(255 255 128);
    --g-color-private-brand-250-solid: rgb(255 255 153);
    --g-color-private-brand-200-solid: rgb(255 255 179);
    --g-color-private-brand-150-solid: rgb(255 255 204);
    --g-color-private-brand-100-solid: rgb(255 255 217);
    --g-color-private-brand-50-solid: rgb(255 255 230);
    --g-color-base-brand: yellow;
    --g-color-base-background: rgb(45, 44, 51);
    --g-color-base-brand-hover: var(--g-color-private-brand-650-solid);
    --g-color-base-selection: var(--g-color-private-brand-150);
    --g-color-base-selection-hover: var(--g-color-private-brand-300);
    --g-color-text-link: var(--g-color-private-brand-600-solid);
    --g-color-text-link-hover: var(--g-color-private-brand-850-solid);
    --g-color-text-brand: var(--g-color-private-brand-600-solid);
    --g-color-text-brand-heavy: var(--g-color-private-brand-850-solid);
    --g-color-line-brand: var(--g-color-private-brand-550-solid);
}

.g-root_theme_dark .yfm {
    --yfm-color-note-info-background: blue;
}

"
`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 4`] = `
"title: Documentation
description: ''
meta:
  title: Documentation
  noIndex: true
links:
  - title: Getting started with Documentation
    description: This guide will show you the basics of working with Documentation
    href: page.md
"
`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 5`] = `
"---
metadata:
  - name: yfm
    content: builder
---

Lorem
"
`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 6`] = `"Lorem"`;

exports[`Build themer feature > Apply theme > md2md with theme yaml 7`] = `
"title: Documentation
href: index.yaml
items:
  - name: Documentation
    href: page.md
  - name: Config
    href: project/config.md
"
`;

exports[`Build themer feature > create css > Create base css 1`] = `
".g-root {
    --g-color-base-brand: rgb(94 33 41);
}

.yfm {
    --yfm-color-note-tip-background: rgb(106, 90, 205);
}

"
`;

exports[`Build themer feature > create css > Create dark css 1`] = `
".g-root_theme_light {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-base-brand: rgb(220, 48, 232);
    --g-color-base-background: rgb(255,255,255);
}

"
`;

exports[`Build themer feature > create css > Create full css 1`] = `
".g-root {
    --g-color-base-brand: rgb(94 33 41);
}

.yfm {
    --yfm-color-note-tip-background: rgb(106, 90, 205);
}

.g-root_theme_light {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-base-brand: rgb(48, 149, 232);
    --g-color-base-background: rgb(255,255,255);
}

.g-root_theme_dark {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-base-brand: rgb(58, 228, 143);
    --g-color-base-background: rgb(45, 44, 51);
}

"
`;

exports[`Build themer feature > create css > Create light css 1`] = `
".g-root_theme_light {
    --g-color-private-brand-50: rgb(94 33 41 / 0.1);
    --g-color-private-brand-100: rgb(94 33 41 / 0.15);
    --g-color-base-brand: rgb(48, 149, 232);
    --g-color-base-background: rgb(255,255,255);
}

"
`;
